### DISCUSSION TITLE
Configuring globals does not appear to work

### DISCUSSION THREAD

**Discussion by tunaranch** [samba-in-kubernetes/sambacc#42] [Q&A]
Hi everyone.

I'm trying to stand up a samba AD DC in minikube for testing, and I can't seem to modify smb.conf to disable the strong auth requirement.

Here's my Kubernetes resources, startup log, and generated smb.conf file

https://gist.github.com/09aed2659f5eeeaa176edd5f4d87e801

Have I made a mistake somewhere?

**Comment by phlogistonjohn** [comment]
> Hi everyone.
> 
> I'm trying to stand up a samba AD DC in minikube for testing, and I can't seem to modify smb.conf to disable the strong auth requirement.
> 

Thanks for trying out our container!  I hope we can resolve this issue.

> Here's my Kubernetes resources, startup log, and generated smb.conf file
> 
> https://gist.github.com/09aed2659f5eeeaa176edd5f4d87e801
> 

Thanks for the info.

> Have I made a mistake somewhere?

Probably not, now that I think about it. Just to confirm, what's missing is the  "ldap server require strong auth": "no"  part right?

This might be a bug in the [sambacc](https://github.com/samba-in-kubernetes/sambacc) layer.  On the fileserver image we import config options using samba's registry configuration backend.  However, the AD DC image does things a bit differently, shelling out to `samba-tool` to create the domain configuration. So it's possible the general options are not getting passed from sambacc's JSON configuration through samba-tool into the conf file.

I will look into the sambacc code when I have a chance to confirm my suspicion. Go ahead and file an issue on the sambacc project too - it'll remind me to investigate - we can always close the issue if my memory is faulty and the issue is not in sambacc. Please note that I may not be able to jump on a fix right away, as the next couple of days as this week is the sambaXP conference and I need to prepare for & present at the conference.  Thanks!

**Reply by tunaranch** [reply]
> Probably not, now that I think about it. Just to confirm, what's missing is the "ldap server require strong auth": "no" part right?

Correct.

> This might be a bug in the [sambacc](https://github.com/samba-in-kubernetes/sambacc) layer. On the fileserver image we import config options using samba's registry configuration backend. However, the AD DC image does things a bit differently, shelling out to samba-tool to create the domain configuration. So it's possible the general options are not getting passed from sambacc's JSON configuration through samba-tool into the conf file.

The same problem happens with the `quay.io/samba.org/samba-server:latest` image as well. I modified the config to be:

```
"configs": {
        "samba-ad": {
          "globals": [ "noprinting", "nossl" ],
          "instance_name": "dc1"
        }
      },
```

and `/etc/samba/smb.conf` looks like

```
[global]
config backend = registry
```

> I will look into the sambacc code when I have a chance to confirm my suspicion. Go ahead and file an issue on the sambacc project too - it'll remind me to investigate - we can always close the issue if my memory is faulty and the issue is not in sambacc. Please note that I may not be able to jump on a fix right away, as the next couple of days as this week is the sambaXP conference and I need to prepare for & present at the conference. Thanks!

I'll do that. Thanks for looking into this, and best of luck for your talk.
